torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```
@@ -373,11 +373,20 @@ bool ProcessGroupNCCL::WorkNCCL::startedGPUExecutionInternal() const {
 }
 
 bool ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const {
-  for (const auto i : c10::irange(devices_.size())) {
-    // Checking the work's corresponding CUDA events' status
-    if (!(*ncclEndEvents_)[i].query()) {
-      return false;
+  try {
+    for (const auto i : c10::irange(devices_.size())) {
+      // Checking the work's corresponding CUDA events' status
+      if (!(*ncclEndEvents_)[i].query()) {
+        return false;
+      }
     }
+  } catch (const std::exception& e) {
+    if (std::string(e.what()).find("driver shutting down") == std::string::npos) {
+      throw;
+    }
+    LOG(INFO) << "[Rank " << rank_
```

Why do we swallow the exception and just continue here? Isn't this misleading to the user where they think the operation has completed but it actually has not?

---

When this happens, the operation has completed so user code is not affected. We just want PG to peacefully exit without aborting.